cmd/node.go

```diff
@@ -273,7 +358,9 @@ func (node *Node) Start(exitChannels ...*chan struct{}) {
 		case <-syscallChannel:
 		}
 
+		node.statusMutex.Lock()
 		node.Stop()
+		node.statusMutex.Unlock()
```

Comment:
Generally I prefer to have the handling of the mutexes confined to the methods that actually interact with the field we're trying to lock on. Could we move the lock/unlock on statusMutex into the `node.Stop` call?

Reply:
> Generally I prefer to have the handling of the mutexes confined to the methods that actually interact with the field we're trying to lock on. Could we move the lock/unlock on statusMutex into the `node.Stop` call?

I'll move these locks inside `node.Stop()`